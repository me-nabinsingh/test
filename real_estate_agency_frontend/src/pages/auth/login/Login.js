import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Formik, Form, Field, ErrorMessage } from 'formik';
import './styles.css';
import Logo from '../../../components/logo/Logo';
import { login } from '../../../services/authService';

class Login extends Component {

    componentWillMount() {
        const token = localStorage.getItem('token');
        if(token) {
            this.props.history.push('/admin');
        }

    }
    render() {
        const { history } = this.props;
        
        return ( 
            <Formik
                initialValues={{ email: '', password: '' }}
                validate={values => {
                    let errors = {};
                    if (!values.email) {
                        errors.email = 'Required';
                    } else if (
                        !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
                    ) {
                        errors.email = 'Invalid email address';
                    }
                    return errors;
                }}
                onSubmit={(values, { setSubmitting }) => {
                    login(values)
                        .then(res => {
                            if(res.data.auth) {
                                localStorage.setItem('token', res.data.token);
                                history.push('/admin');
                            }
                            setSubmitting(false);
                        })
                        .catch(function (error) {
                            console.log(error);
                            setSubmitting(false);
                        });
                }}
                >
                {({ isSubmitting }) => (
                    <Form className="form-signin text-center" >
                    <div className="my-5">
                        <Logo />
                    </div>
                    <h1 className="h3 mb-3 font-weight-normal">Admin Log In</h1>
                    <label htmlFor="inputEmail" className="sr-only">Email address</label>
                    <Field className="form-control" type="text" name="email" />
                    <ErrorMessage className="text-danger" name="email" component="div" />

                    <label htmlFor="inputPassword" className="sr-only">Password</label>

                    <Field className="form-control" type="password" name="password" />
                    <ErrorMessage name="password" component="div" />
                    <button type="submit" className="btn btn-lg btn-primary btn-block" disabled={isSubmitting}>
                        Submit
                    </button>

                    <div className="mt-3">
                        <Link to="/auth/register">Register</Link>
                    </div>
                    </Form>
                )}
            </Formik>
         );
    }
}
 
export default Login;