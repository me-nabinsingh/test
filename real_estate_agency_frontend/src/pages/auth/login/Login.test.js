import React from 'react';
import Login from './Login';
import renderer from 'react-test-renderer';
import {Router } from "react-router-dom";
import history from "../../../util/history";

describe('<Login />', () => {
    it('renders without crashing', () => {
        const component = renderer.create(
            <Router history={history}>
                <Login />
            </Router>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});