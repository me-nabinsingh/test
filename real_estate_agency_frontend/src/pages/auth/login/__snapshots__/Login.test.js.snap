// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders without crashing 1`] = `
<form
  className="form-signin text-center"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="my-5"
  >
    <a
      className="py-2"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        height={50}
        src="logo.svg"
        width={50}
      />
    </a>
  </div>
  <h1
    className="h3 mb-3 font-weight-normal"
  >
    Admin Log In
  </h1>
  <label
    className="sr-only"
    htmlFor="inputEmail"
  >
    Email address
  </label>
  <input
    className="form-control"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="sr-only"
    htmlFor="inputPassword"
  >
    Password
  </label>
  <input
    className="form-control"
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    className="btn btn-lg btn-primary btn-block"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
  <div
    className="mt-3"
  >
    <a
      href="/auth/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</form>
`;
